doc/perf.md

```diff
@@ -274,9 +260,8 @@ Aborted
 ```
 
 Accessing the value of a NULL pointer is an example of **undefined behavior** (UB).
-The optimizer assumes that your program does not have UB,
-and if your program does something that is UB,
-it could in principle do anything when it is ran with optimizations enabled.
+The optimizer is set to assume that your program does not have anything that causes UB,
```

Comment:
```suggestion
The optimizer assumes that your program does not have anything that causes UB,
```

Reply:
I wouldn't humanize it too much :D

Reply:
"is set" is misleading. It's as if there was some setting inside the optimizer. There isn't. Optimizers always make these assumptions by design.

Reply:
The optimizer -naturally- assumes that your program does not have anything that causes UB,

Reply:
Seems good to me. Here's a suggestion that does exactly that:

```suggestion
The optimizer naturally assumes that your program does not have anything that causes UB,
```

Reply:
"naturally" feels a bit weird to me. How about:

```suggestion
By desing, the optimizer assumes that your program does not have anything that causes UB,
```